programs/drift/src/controller/liquidation.rs

```diff
@@ -1888,6 +1910,18 @@ pub fn liquidate_perp_pnl_for_deposit(
         return Err(ErrorCode::InvalidLiquidation);
     }
 
+    if safest_tier_perp_liability < contract_tier
+        || safest_tier_spot_liability < AssetTier::default()
+    {
+        msg!(
+            "liquidating contract tier={:?} is riskier than outstanding {:?} & {:?}",
+            contract_tier,
+            safest_tier_perp_liability,
+            safest_tier_spot_liability
+        );
+        return Err(ErrorCode::TierViolationLiquidatingPerpPnl);
```

Comment:
In case somebody calls this ix and it leads to the users orders to be canceled, we might just want to return early instead of erring?